api/app/src/main/kotlin/packit/service/PacketService.kt

```diff
@@ -85,6 +82,15 @@ class BasePacketService(
             .orElseThrow { PackitException("packetNotFound", HttpStatus.NOT_FOUND) }
     }
 
+    override fun getDependencies(id: String): List<Packet> {
+        val dependencyIds = getMetadataBy(id).depends.map() { it.packet }
+        return packetRepository.findAllById(dependencyIds)
+    }
```

Comment:
this may not be needed. as if we cant really trust the dependencies what we get back from the api, it may make sense to update the `getMetadataBy` and then update the depends field to be a List<`BasePacketDto`>....

Reply:
> if we cant really trust the dependencies what we get back from the api

Why can't we trust them?

Reply:
Oh do you mean that I'm deriving the dependencies from the proxy db, which might be out of date?